tests/unit/test.config.js

```diff
@@ -460,10 +545,39 @@ describe('config', () => {
         argv,
         configObject,
         defaultValues,
+        optionTypes,
         commandExecuted,
       });
       assert.strictEqual(newArgv.apiKey, cmdApiKey);
     });
+
+    it('throws an error when the type of sub option is invalid', () => {
```

Comment:
This needs another test that configures a value for a command that is not being executed. This would highlight the error you had where the `optionTypes` are not keyed by command. For example, set up a test where the user executes `run` but specifies an invalid type in a config file for `sign.apiKey`.